lib/actions/fcs.py

```diff
@@ -339,6 +366,422 @@ def fcs2html(self, req):
                     fcs_custom_hdr_inject=custom_hdr_inject)
 
 
+    def _get_element_tree(self, ns, title, data, attributes={}):
+        """
+        Creates xml.etree.ElementTree object from given data.
+        :param ns: str, namespace
+        :param title: str, root node title
+        :param data: list, elements of different types
+        :param attributes: dict, root node attributes,
+        :return: xml.etree.ElementTree, created object
+        """
+        if not ns.endswith(':'):
+            ns += ':'
+
+        root = ET.Element(ns + title)
+        root.attrib = attributes
+
+        for elem in data:
+            if elem is None:
+                continue
+
+            if type(elem) == ET.Element:
+                root.insert(MAX_INT, elem)
+
+            else:
+                if len(elem) == 2:
+                    attributes = {}
+                    key, value = elem
+                else:
+                    key, value, attributes = elem
+
+                if type(value) == list:
+                    el = self._get_element_tree(ns, key, value, attributes)
+                    root.insert(MAX_INT, el)
+
+                else:
+                    el = ET.SubElement(root, ns + key)
+                    el.text = value
+                    el.attrib = attributes
+
+        return root
+
+    def _get_endpoint_description_tree(self, parameters, ns='ed'):
+        """
+        Creates xml.etree.ElementTree object for fcs endpoint description site.
+        :param parameters: dict
+        :param ns: str, namespace
+        :return: xml.etree.ElementTree
+        """
+        default_corpus_info = self._corpus_info(parameters['corpname'])
```

Comment:
Shouldn't resources contain all the available corpora and not just the default one?

Reply:
As of now, only default corpora is available through fcs